postgres-appliance/bootstrap/maybe_pg_upgrade.py

```diff
@@ -61,6 +65,11 @@ def main():
         raise Exception('Failed to upgrade cluster from {0} to {1}'.format(cluster_version, bin_version))
 
     logger.info('Starting the cluster with new postgres after upgrade')
+    if archive_mode is None:
+        postgresql_config['parameters'].pop('archive_mode')
```

Comment:
didn't you pop this already in line 14?

Reply:
line 17 sets it: https://github.com/zalando/spilo/pull/331/files#diff-70677ed4eab8d8893711a96cb7a8580fR17